### PR TITLE
B 18310 multi move feature flag

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary

This PR introduces the frontend Multi Move feature flag, the enabling of the experimental environment-variable-based feature flagging for the backend called EnvFetcher, and the backend environment variable setup for Multi Move.

### This PR proposes the following:
Multi Move to be an enabled feature inside of `development` and `experimental`. It will be disabled in all other environments.

## Documentation
The EnvFetcher is the "old" method of utilizing feature flags within MilMove. Documentation and the existing codebase points to utilizing "FeatureFlagFetcher" which is configured for Flipt. Flipt was never fully enabled so please disregard these findings. Be aware of this when proceeding with feature flag work.

EnvFetcher was previously undocumented. Please see documentation changes introduced [here](https://github.com/transcom/mymove-docs/pull/462). This documentation includes specifically how to utilize this new feature flag.

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. This is not inherently testable on its own. If you do want to test it, you have to make a local code change.
2. In your development environment of this branch, run `git revert b1c421ab535843d7516e276ca90d9aec9eeca741`, then run the client to see the frontend feature flag.
3. For the backend, please refer to the documentation.

### CODE REVIEW

**Review by TevinAdams — APPROVED** [review]

File: pkg/services/feature_flag.go
Comment (traskowskycaci) [inline pkg/services/feature_flag.go:1]: Do you want the //go:generate mockery --name EnvFetcher line left in there?

**Review by traskowskycaci — APPROVED** [review]
Looks good, left one question in about a comment

**Comment by cameroncaci** [comment]
Still need one more review on the documentation portion of the PR: https://github.com/transcom/mymove-docs/pull/462

**Review by danieljordan-caci — APPROVED** [review]
LGTM!

**Pull request merged**